Домашнє завдання №1

Ви —  продакт менеджер в компанії, яка займається мобільними додатками. Ваша модель монетизації — підписка. У травні 2019 року до вас приєдналося 100 підписників, у червні — 150, у липні — 170. Скільки ці користувачі принесли вам до початку травня 2021 року (за два роки), якщо щомісячний рівень відтоку становить 5%, а підписка коштує $10?

S1 05.2019 = 100 підписників (24 місяця)
S2 06.2019 = 150 підписників (23 місяця)
S3 07.2019 = 170 підписників (22 місяця)

Cr (Churn rate) = 5% = 0.05
Price  = 10 $
R(totall) = S1 + S2 + S3
R(S1) = price * S1 * (1-q**n) / (1-q)

In [1]:
S1 = 100       # Перша група підписників
price = 10     # Ціна підписки
q = 0.95       # Рівень утримання
n = 24         # Кількість місяців

R1 = price * S1 * (1 - q**n) / (1 - q)
print(R1)

14160.21951322454


In [3]:
S2 = 150       # Друга група підписників
price = 10     # Ціна підписки
q = 0.95       # Рівень утримання
n = 23         # Кількість місяців

R2 = price * S2 * (1 - q**n) / (1 - q)
print(R2)

20779.293968249276


In [5]:
S3 = 170       # Третя група підписників
price = 10     # Ціна підписки
q = 0.95       # Рівень утримання
n = 22         # Кількість місяців

R3 = price * S3 * (1 - q**n) / (1 - q)
print(R3)

22999.859470893873


In [7]:
R = round(R1 + R2 + R3, 2)
print(R)

57939.37


Відповідь: загальний дохід від усіх користувачів за заданий період складає 57939,37$

2. Ви — менеджер по роботі з продуктом в компанії, яка розробляє мобільні ігри. У січні 2020 року у вас було 2000 активних користувачів. Щомісяця кількість нових користувачів зростає на 3%, але водночас 2% існуючих користувачів перестають користуватися додатком. У результаті, ефективне зростання активних користувачів складає 1% щомісяця.
    
    Ваша мета — визначити кумулятивну кількість активних користувачів за весь період від січня 2020 року до кінця грудня 2021 року (тобто за 24 місяці), враховуючи щомісячний приріст 1%.

Ми використовуємо формулу для суми геометричної прогресії:
R = S1 * (1-q**n) / (1-q)
R — сумарна кількість активних користувачів за весь період.
S1 = 2000 — кількість користувачів на початок періоду.
q = 1.01 — коефіцієнт приросту.
n = 24 — кількість місяців.

In [21]:
S1 = 2000
q = 1.01
n = 24
R = round(S1 * (1-q**n) / (1-q), 2)
print(R)

53946.93


Відповідь: кумулятивна кількість активних користувачів за весь період від січня 2020 року до кінця грудня 2021 року приблизно 53947 користувачів.

3. Ви — менеджер з розвитку бізнесу в компанії, що займається створенням платформи для онлайн-курсів. У січні 2021 року ви залучили 500 нових користувачів. У лютому кількість нових користувачів зросла на 50%, у березні — ще на 20%. З квітня приріст нових користувачів стабілізувався на рівні 10% щомісяця. Скільки активних користувачів ви мали загалом за перше півріччя 2021 року?

S1 01.2021 = 500 користувачів.
S2 02.2021 = 500 підписників + 50% = 750 користувачів.
S3 03.2021 = 750 підписників + 20% = 900 користувачів.
Для наступного періоду використовуємо формулу для суми геометричної прогресії:
R = S3 * (1-q**n) / (1-q)
R — сумарна кількість активних користувачів за період (квітень, травень, червень).
S3 = 900 — кількість користувачів на початок періоду.
q = 1.1 — коефіцієнт приросту на період (квітень, травень, червень).
n = 3 — кількість місяців.
R = 900 * (1 - 1.1**3) / (1 - 1.1) = 2979 (кількість активних користувачів за період (квітень, травень, червень)
S(total) = S1 + S2 + S3 + R
S(total) = 500 + 750 + 900 + 2979 = 5129 активних користувачів ми отримали загалом за перше півріччя 2021 року

In [23]:
S3 = 900
q = 1.1
n = 3
R = round(S1 * (1-q**n) / (1-q), 2)
print(R)

2979.0


Відповідь: 5129 активних користувачів ми отримали загалом за перше півріччя 2021 року.